.idea/libraries/Dart_practice/chapter_four.dart

```diff
@@ -0,0 +1,233 @@
+
+  const myAge = 33;
+  const isTeenager = myAge > 13 && myAge < 19;
+  print("I am a teenager? $isTeenager");
+  print("---------------------------");
+
+  // 1. “Create another constant named maryAge and set it to 30.”
+  // 2. “Then, create a constant named bothTeenagers that uses Boolean logic to determine if both you and Mary are teenagers.”
+
+  const maryAge = 30;
+  const bothTeenagers = (maryAge > 13 && maryAge < 19) && isTeenager;
+  print(bothTeenagers);
+  print("---------------------------");
+
+  // 1. “Create a String constant named reader and set it to your name.”
+  // 2. “Create another String constant named ray and set it to 'Ray Wenderlich'.”
+  // 3. “Create a Boolean constant named rayIsReader that uses string equality to determine if reader and ray are equal."
+
+  const reader = "Leo";
+  const ray = "Ray Wenderlich";
+  const rayIsReader = reader == ray;
+  print("Equality? $rayIsReader.");
+  print("---------------------------");
+
+  // 1. “Create a constant named myAge and initialize it with your age.”
+  // 2. “Write an if statement to print out “Teenager” if your age is between 13 and 19, and “Not a teenager” if your age is not between 13 and 19.”
+
+  const myAgeAgain = 33;
+  if (myAgeAgain > 13 && myAgeAgain < 19) {
+    print("Teenager.");
+  } else {
+    print("Not a teenager.");
+  }
+  print("---------------------------");
+
+  // 1. “Use a ternary conditional operator to replace the else-if statement that you used above. Set the result to a variable named answer.”
+
+  const teenager = (myAgeAgain > 13 && myAgeAgain < 19)
+      ? "Teenager"
+      : "Not a teenager.";
+  print(teenager);
+  print("---------------------------");
+
+  // 1. “Make an enum called AudioState and give it values to represent playing, paused and stopped states.”
+  // >>> see above;
+  // 2. “Create a constant called audioState and give it an AudioState value. Write a switch statement that prints a message based on the value.”
+
+  const audioState = AudioState.playing;
+  switch (audioState) {
+    case AudioState.playing:
+      print("▶️");
+      break;
+    case AudioState.paused:
+      print("⏸️");
+      break;
+    case AudioState.stopped:
+      print("⏹️");
+      break;
+  }
+  print("---------------------------");
+
+  // 1. “Create a variable named counter and set it equal to 0. Create a while loop with the condition counter < 10.”
+  // 2. “The loop body should print out “counter is X” (where X is replaced with the value of counter) and then increment counter by 1.”
+
+  var counter = 0;
+  while (counter < 10) {
+    print("Counter is $counter.");
+    counter += 1;
+  }
+  print("---------------------------");
+
+  // 1. “Write a for loop starting at 1 and ending with 10 inclusive. Print the square of each number.”
+
+  for (var i = 1; i <= 10; i++) {
+    var square = pow(i, 2);
```

Comment:
@octavianChiriches sqrt(i) calculates the square root, pow returns I to the power of 2 :P